docs/reference/nodes.md

```diff
@@ -1,5 +1,68 @@
 # Shift Nodes
 
-## 1.0.0
+Nodes are the graphic units that build up the workflow. Each node is associated to an [operator](../basics/terminology/#operator), which defines the execution logic of the node, and are connected to other nodes through [plugs](../basics/terminology/#plug).
 
-To Be Defined
\ No newline at end of file
+## Node Interface
+
+There are four main interface elements that compose the node: the node's name, the operator type, the trigger plugs and the i/o's plugs.
+
+
+![Node UI](../../images/node_ui.png "Node UI")  
+
+* **Node Name**: Defines the name of the node. On creation, the name of the node is the name of the operator with an enumerated suffix. The name of the node can be changed through the [*Inspector Widget*](../getting_started/basics/ui_overview/#inspector).
+* **Operator Type**: The associated operator to the node. It determines the native plugs of the node.
+* **Trigger Plugs**: The trigger plugs (`>>>`) are plugs meant to define the node's position in the execution graph. They do not pass any data. If the node has other input or output plugs connected to another node, the trigger plugs do not need to be connected; otherwise, it is recommended to connect these plugs to indicate when the node needs to get computed.
+
+* **I/O's Plugs**: Input and output plugs are define the incoming data required by the operator and the resulting data after it is computed. 
+
+### Inspect a Node
+To read further information about a node, `Right-Click` on the node and `Click` on the *Help* option. This will prompt a dialog with the following information:
+
+* The node's name.
+* The operator type and the [Catalog](../reference/catalogs/#catalogs) it belongs to.
+* The operator description.
+* The input and output plugs with their corresponding plug types.
+
+![Shift Node Information Dialog](../images/shift_operator_dialog.gif)
+
+## Plugs
+Plugs define the input and output data of the nodes. Data values can be set through the [*Inspector Widget*](../getting_started/basics/ui_overview/#inspector) or by connecting plugs, which allows nodes to pass data from one to another. The kind of data compatible with a plug is defined by the plug type.
+
+**Plug Types**
+
+| Type | Description | Compatible Python Type |
+| :--- | :---------- | :---|
+| *Bool* | Defines a boolean value.|`Bool`|
+| *Code* | Defines a python code block. This type is used whenever custom scripting is required in a node. A plug of this type displays as an embedded text editor in the inspector. |`String`|
+| *Color* | Defines a color in RGBA. Interacting with a plug of this type opens a color picker dialog.|`List`|
+| *Dict* | Defines a dictionary value.|`Dict`|
+

```

Comment:
I think you need to remove this line, the table does not display nicely
![image](https://github.com/user-attachments/assets/0e5ce4d3-f075-443b-80f7-741159158d99)
```suggestion

```

Reply:
Done